ci/serverless/common.sh

```diff
@@ -12,11 +12,12 @@ setup_vault() {
   vault_path=secret/ci/elastic-logstash/serverless-test
   set +x
   export ES_ENDPOINT=$(vault read -field=es_host "${vault_path}")
-  export ES_USER=$(vault read -field=es_user "${vault_path}")
-  export ES_PW=$(vault read -field=es_user_pw "${vault_path}")
+  export ES_USER=$(vault read -field=es_superuser "${vault_path}")
```

Comment:
can we get rid of the user/password secrets and just use the api keys ? I see they're still used here:

```
ci/serverless/elastic_integration_filter_tests.sh:  PIPELINE_RESP_CODE=$(curl -s -w "%{http_code}" -o /dev/null -X PUT -u "$ES_USER:$ES_PW" "$ES_ENDPOINT/_ingest/pipeline/integration-logstash_test.events-default" \
ci/serverless/elastic_integration_filter_tests.sh:  TEMPLATE_RESP_CODE=$(curl -s -w "%{http_code}" -o /dev/null -X PUT -u "$ES_USER:$ES_PW" "$ES_ENDPOINT/_index_template/logs-serverless-default-template" \
```

Are the permissions of the "logstash_user" key enough for this too?

Reply:
No. The key is not enough

Reply:
removed super user. I will address the minimum privilege in other PR and create a new api_key with `manage_ingest_pipelines` for data preparation.